### PR TITLE
6546 contact info and extra functionality popover

### PR DESCRIPTION
**Pull request by TeroOtavia**
Info popover to show data when hovering activator element (Example recipient/sender name in communicator). Branch also ads nameExtension to workspace name that is either showed as recipient.

Needs Pyramus 6546_contact_info_popover

Resolves: #6546 
Resolves: #6555

### CODE REVIEW

**Review by avukkonen — DISMISSED** [review]

**Review by pkukkon — CHANGES_REQUESTED** [review]

**Review by aviljakaine — CHANGES_REQUESTED** [review]
`loggedUserHasPermission` could've maybe been named better to indicate what it means :D

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/guider-link.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+   */
+  const onSubmit = () => {
+    const url = `https://${window.location.hostname}/guider#?c=${schoolDataIdentifier}`;
+
+    window.open(url);
+  };
+
+  return (
+    <Link
+      aria-label="Avaa opiskelijan ohjausnäkymä"
```
Comment (avukkonen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/guider-link.tsx:37]: Todo: lokalisointi

**Review by avukkonen — CHANGES_REQUESTED** [review]

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/schooldatalocal/LocalUserSchoolDataBridge.java [hunk 2]
```diff
@@ -692,5 +692,4 @@ public boolean amICounselor(String studentIdentifier) {
   public List<UserStudyPeriod> listStudentStudyPeriods(SchoolDataIdentifier userIdentifier) {
     throw new SchoolDataBridgeInternalException("Not supported");
   }
-
```
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/schooldatalocal/LocalUserSchoolDataBridge.java:695]: The whole file can be reset to HEAD if it's just linebreaks. Same with `UserSchoolDataBridge` and `PyramusUserSchoolDataBridge`

File: muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserRESTService.java [hunk 3]
```diff
@@ -295,6 +297,159 @@ public Response setUserEntityProperty(fi.otavanopisto.muikku.rest.model.UserEnti
...
+        if (d.equals(UserInfo.AVATAR)) {
+          Boolean hasAvatar = userEntityFileController.hasProfilePicture(userEntity);
+          result.put("hasAvatar", hasAvatar.toString());
+        }
+        
+        if (result.get("loggedUserHasPermission").equals("false") && !loggedUserRole.equals(EnvironmentRoleArchetype.STUDENT)) {
+          return Response.ok(result).build();
+        }
+        
+        if (d.equals(UserInfo.EMAIL)) {
```
[the remark this answers is not in the document]
Comment (pkukkon) [reply muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserRESTService.java:375]: Student vs. student resolves to 403 earlier, so this scenario shouldn't happen.

**Review by pkukkon — CHANGES_REQUESTED** [review]

**Review by pkukkon — CHANGES_REQUESTED** [review]
Some minor logic changes based on the unresolved comments.

**Review by pkukkon — APPROVED** [review]

**Review by aviljakaine — APPROVED** [review]

**Pull request merged**